src/executor_utils.c

```diff
@@ -0,0 +1,155 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   executor_utils.c                                   :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: sqiu <sqiu@student.42vienna.com>           +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/08/01 18:03:04 by sqiu              #+#    #+#             */
+/*   Updated: 2023/08/08 14:52:48 by sqiu             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+/**
+ * @file executor_utils.c
+ * @brief Helper functions for executor.
+ */
+
+#include "mod_executor.h"
+
+/**
+ * @brief Check given args executable.
+ * 
+ * If args executable contains an absolute path (starts with /),
+ * a relative path (starts with ./ or ../) or no paths are given,
+ * its validity is checked in the current directory. If not successfully,
+ * ERR_UNKNOWN_CMD is returned.
+ * 
+ * Else an attempt to find and assign the right path is started. If not
+ * successfully, ERR_UNKNOWN_CMD is returned.  
+ * @param args 			String array containing executable in first position.
+ * @param cmd_paths 	String array of paths.
+ * @return t_err 		ERR_UNKNOWN_CMD, ERR_MALLOC, SUCCESS
+ */
+t_err	ft_check_cmd_access(char **args, char **cmd_paths)
+{
+	t_err	err;
+
+	err = ERR_UNKNOWN_CMD;
+	if ((*args)[0] == '/' || !ft_strncmp(*args, "./", 2)
+		|| !ft_strncmp(*args, "../", 3) || !cmd_paths)
+	{
+		if (access(*args, F_OK | X_OK) == 0)
```

Comment:
Not sure we need to catch that "error" because when it's returning -1, it basically is just saying that it didn't find the right path+cmd combination yet. That's why it's looping through all paths. When rly no good combo is found, then the according error message is printed by ft_process_cmd()

Reply:
Depends on how "deep" we want to go into error handling: access (or nearly all syscalls) set different errno values depending on the encountered error. We could wrap to catch errno and print the corresponding errno message with perror(). 
I'll add a "maybe" issue